### DISCUSSION TITLE
Using @babel/standalone witn Next.js and got "Module not found: Can't resolve 'fs'"

### DISCUSSION THREAD

**Discussion by tomitrescak** [babel/babel#13630] [Q&A]
Hi, for some reason I cannot use @babel/standalone with "next.js". I want to compile things in the browser, but I'm getting the `Can't resolve 'fs'` error.

this is my code:

```ts
import * as Babel from '@babel/standalone';

Babel.registerPlugin(
  '@babel/plugin-proposal-class-properties',
  require('@babel/plugin-proposal-class-properties')
);

const standardConfig = {
  filename: 'example.ts',
  presets: ['typescript'],
  plugins: ['@babel/plugin-proposal-class-properties']
};

export function compile(code: string) {
    return Babel.transform(
          code,
          standardConfig 
    ).code;
}
```

Do you have any idea what am I doing wrong?
Thanks!

This is the error:

```
error - ./node_modules/@babel/core/lib/transformation/normalize-file.js:9:0
Module not found: Can't resolve 'fs'
```

**Comment by tomitrescak** [comment]
It was the "@babel/plugin-proposal-class-properties" problem ...
The change necessary was:

```
const standardConfig = {
  filename: 'example.ts',
  presets: ['typescript'],
  plugins: ['proposal-class-properties']
};
```

**Comment by JLHwung** [comment]
This is expected. '@babel/plugin-proposal-class-properties' is a node package and is not supposed to work in browser. If you are using `@babel/standalone`, you can view available plugins programmatically by `Babel.availablePlugins` or checkout the [source](https://github.com/babel/babel/blob/main/packages/babel-standalone/src/generated/plugins.ts).
